### ISSUE TITLE
a version of the semantic segmentation task for regression

### ISSUE THREAD

**Issue by Geethen** [microsoft/torchgeo#849]
### Summary

semantic segmentation models does support regression if 'classes' are changed to 1 and the 'activation' is changed to 'identity'. it will be useful to have a task created for this in torchgeo.

Also, it would be useful if segmentation models supported more recent architectures (not a torchgeo problem)- out of my depth for me to implement.

**Comment by adamjstewart** [comment]
@isaaccorley this may be useful for our GHG project.

**Comment by isaaccorley** [comment]
> @isaaccorley this may be useful for our GHG project.

I actually have a draft of this regression trainer which uses MSE loss. I'll make a PR soon.

**Comment by isaaccorley** [comment]
@Geethen Can you list a few architectures you would like to see available in the segmentation trainer? From experience, vanilla UNet with some training tricks is generally hard to beat which is why we chose segmentation models pytorch as the backend.

**Comment by Geethen** [comment]
@isaaccorley I have heard this before as well. Nevertheless, it will be cool to have some of these recent models available [segnext](https://arxiv.org/abs/2209.08575), [stego](https://mhamilton.net/stego.html) there are lots of stuff in this repo: https://github.com/robmarkcole/satellite-image-deep-learning#Segmentation.

two models with a proven record (tend to always be some adaptation of UNet as you mentioned): 
global tree cover layer: https://github.com/wri/sentinel-tree-cover
Google dynamic world: https://github.com/google/dynamicworld

**Comment by calebrob6** [comment]
@Geethen, btw the google dynamic world paper uses a simple full convolutional net I think the global tree cover layer also uses a simple fcn (but takes in a time series and has recurrent layers)
